extensions/ql-vscode/src/local-databases.ts

```diff
@@ -897,6 +897,20 @@ export class DatabaseManager extends DisposableObject {
     }
   }
 
+  public async digForDatabaseItem(
+    language: string,
+    databaseNwo: string,
+  ): Promise<DatabaseItem | undefined> {
+    const dbItems = this.databaseItems || [];
+    const dbs = dbItems.filter(
+      (db) => db.language === language && db.name === databaseNwo,
+    );
+    if (dbs.length === 0) {
+      return undefined;
+    }
+    return dbs[0];
+  }
```

Comment:
I initially started out with defining the method in the wizard, but the `databaseItems` property on the `databaseManager` is private & readonly so when trying to pass it in, I'd get this error:

```
Argument of type 'readonly DatabaseItem[]' is not assignable to parameter of type 'DatabaseItem[]'.
  The type 'readonly DatabaseItem[]' is 'readonly' and cannot be assigned to the mutable type 'DatabaseItem[]'.
```

I agree that class is massive, but on the other hand I'm also calling `setCurrentDatabaseItem` on it as well, so I think it makes sense for it to deal with databaseItem logic. Also there are two other search methods in that class: `findDatabaseItem` (searches by Uri) and `findDatabaseItemBySourceArchive`.

Reply:
We could remove the `readonly` on the attribute, but I think it would be cooler if we moved that large class into its own file to make it more scannable. I'm happy to look at that in a separate PR.

Reply:
You should be able to change the type from `databaseItems: DatabaseItem[]` to `databaseItems: readonly DatabaseItem[]` since you aren't modifying the array or items. If that doesn't work, there's also `databaseItems: (readonly DatabaseItem)[]` which just sets the `readonly` on the items instead of on the array.

I'm not sure if I mentioned this in my initial review, but I think the naming of this method could also be clearer, something like `findDatabaseItemByNwo`.

Reply:
Yeah the naming is definitely a thing I struggled with. Cool, I can give it a shot (I didn't know you could do that! I can't think of a case where this would be a bad idea.)

Reply:
I'm going to settle for moving the search methods in the skeleton wizard file, because I'm not sure where helper methods like these would go if we had to define a separate file. Open to suggestions, if you have any ideas.

Reply:
I've done the refactoring in a separate PR: https://github.com/github/vscode-codeql/pull/2292